src/app/component/shared/components/form-base/form-base.component.ts

```diff
@@ -60,17 +59,9 @@ export class FormBaseComponent implements ComponentCanDeactivate {
   public checkChanges(): boolean {
     const body = this.getFormValues();
     for (const key of Object.keys(body)) {
-      if (Array.isArray(body[key])) {
-        if (body[key].some((item, index) => item !== this.initialValues[key][index])) {
-          return true;
-        }
-      } else {
-        if (body[key] !== this.initialValues[key]
-          && this.formEditProf.nativeElement.classList.contains('ng-touched')) {
-          return true;
-        }
+      if (JSON.stringify(body[key]) !== JSON.stringify(this.initialValues[key]) && this.initialValues[key] !== undefined) {
```

Comment:
check falsy value like this: !this.initialValues[key]

Reply:
https://www.freecodecamp.org/news/falsy-values-in-javascript/#:~:text=A%20falsy%20value%20is%20something,)%2C%20and%20false%20of%20course.

Reply:
I've checked both possible ways, but none of them will fix the bug as that code that I already wrote